src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/FileEmbeddingProviderConfig.java

```diff
@@ -0,0 +1,15 @@
+package io.stargate.sgv2.jsonapi.service.embedding.configuration;
+
+import io.smallrye.config.ConfigMapping;
+import jakarta.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * Do NOT inject this interface, use EmbeddingProvidersConfig instead EmbeddingProvidersConfig bean
+ * produced by EmbeddingProvidersConfigProducer based on EGW enabled or not
+ */
+@ConfigMapping(prefix = "stargate.jsonapi.embedding")
+public interface FileEmbeddingProviderConfig {
+  @Nullable
```

Comment:
I think `DefaultEmbeddingProviderConfig` is better than `FileEmbeddingProviderConfig`

Reply:
Have thought about that, but IMO setting File as prefix for this class is more readable?

Reply:
OK, second thought, change it into defaultEmbeddingProviderConfig